internal/model/modelindexer/indexer.go

```diff
@@ -308,6 +309,17 @@ func encodeAny(v interface{}, out *fastjson.Writer) error {
 		return encodeMap(v, out)
 	case map[string]interface{}:
 		return encodeMap(v, out)
+	case float64:
+		// truncate the value with int conversion to
+		// check for decimals.
+		if v == float64(int64(v)) {
+			// we cannot rely on fastjson.Marshal as it would strip
+			// the leading zero and encode the value as an "int".
+			// So we call AppendFloat directly with precision=1.
+			out.RawBytes(strconv.AppendFloat(nil, v, 'f', 1, 64))
+			return nil
+		}
+		return fastjson.Marshal(out, v)
```

Comment:
```suggestion
```

This should not be necessary. We'll map it as a double even if it's encoded as an integer, by specifying the dynamic template by name.

Reply:
The system test is failing without this line. Am I missing something ?

Reply:
I overlooked something: I was thinking that all metrics would have an entry in the "_metric_descriptions" map which is used in the ingest processor for specifying dynamic templates. This isn't the case: we only add entries for metrics with either Type or Unit set. So we'll need some more changes:

- Always send a metric description, even if the type and unit are empty. This would enable us to identify metric fields in the ingest pipeline, and assign them the correct dynamic template. The changes would need to be made in https://github.com/elastic/apm-server/blob/a1d1cfb3e07dbf5e6ab95e5f1b7a95faddfbd374/internal/model/metricset.go#L163-L172
- Update the `internal_metrics` data stream's ingest pipeline to discard metric descriptions, since in that data stream we do not use dynamic mapping at all.

I'm open to alternative ideas.

Reply:
Sorry for the long wait 🙇 

I learned quite a lot while working on this. I've pushed some changes!